python-lib/config.py

```diff
@@ -84,19 +84,21 @@ def get_hdfs_location(dataset_config: Mapping[AnyStr, Any], sf_stage_name: AnySt
 
 def get_table_schema_sql(sf_table_name: AnyStr) -> AnyStr:
     """
-    Generates SQL to extract table columns from DB
-    :param sf_table_name: Snowflake table in format of "schema"."table". Note that schema and table are case-sensitive.
+    Generates SQL to extract a table's schema
+    :param sf_table_name: Snowflake table in format of "schema"."table"
     :return: SQL for getting table columns
     """
     tab = sf_table_name.replace('"', '')
     schema = tab.split('.')[0]
     table = tab.split('.')[1]
 
-    sql = f'''
-SELECT column_name AS "name", data_type AS "originalType"
+    # Dataiku's `originalType` parameter is Snowflake's data_type but without `_`
+    sql = f"""
+SELECT column_name AS "name", REPLACE(data_type, '_', '') AS "originalType"
```

Comment:
What is the purpose for this change? This is going to break my PR, so not sure if this is correct.

Reply:
This is instead of your PR, not in addition to.

Reply:
Fair enough. Still I see no purpose for it ...

Reply:
If you aim to produce Dataiku datatypes from Snowflake datatypes - the code is incomplete. You need to implement mapping for other datatypes as well. Otherwise results of this method are not reliable.
`REPLACE(data_type, '_', '')` does not handle all datatypes imho.

Reply:
I'm not sure what you mean by this—there are no filters on `data_type` so all are mapped. Do you have an example?